src/cpp/fastdds/log/Log.cpp

```diff
@@ -331,13 +331,16 @@ void Log::get_timestamp(
 #if defined(_WIN32)
     struct tm timeinfo;
     localtime_s(&timeinfo, &now_c);
-    stream << std::put_time(&timeinfo, "%F %T") << "." << std::setw(3) << std::setfill('0') << ms << " ";
     //#elif defined(__clang__) && !defined(std::put_time) // TODO arm64 doesn't seem to support std::put_time
     //    (void)now_c;
     //    (void)ms;
+#elif defined(__unix__)
```

Comment:
According to ctime(3) manpage there's an additional define that should be checked:

`Feature Test Macro Requirements for glibc (see feature_test_macros(7)):

       asctime_r(), ctime_r(), gmtime_r(), localtime_r():
              _POSIX_C_SOURCE
                  || /* Glibc versions <= 2.19: */ _BSD_SOURCE || _SVID_SOURCE`

Reply:
Addressed in 66c46be